pkg/receive/limiter.go

```diff
@@ -77,27 +77,29 @@ func (l *Limiter) HeadSeriesLimiter() headSeriesLimiter {
 	return l.headSeriesLimiter
 }
 
-func (l *Limiter) ShouldRejectNewRequest() bool {
+func (l *Limiter) ShouldRejectNewRequest() (bool, string) {
 	// maxPendingRequests doesn't change once set when a limiter instance is created.
 	// So, it's safe to read it without a lock.
-	if l.maxPendingRequests > 0 && l.pendingRequests.Load() >= l.maxPendingRequests {
-		if l.maxPendingRequestLimitHit != nil {
-			l.maxPendingRequestLimitHit.Inc()
+	if l.maxPendingRequests > 0 {
+		if pendingRequests := l.pendingRequests.Load(); pendingRequests >= l.maxPendingRequests {
+			if l.maxPendingRequestLimitHit != nil {
+				l.maxPendingRequestLimitHit.Inc()
+			}
+			if l.pendingRequestsGauge != nil {
```

Comment:
i think tracking pending request should always on, check nil seems unnecessary and hurts the code readability

Reply:
That's how existing instrumentation works. I believe there is a code path such that those metrics are not initiated.
<img width="745" alt="image" src="https://github.com/user-attachments/assets/f7b3fca6-0d5f-445a-8ecc-5a649b2eb638" />

<img width="657" alt="image" src="https://github.com/user-attachments/assets/6b8fe703-a3ad-4504-9a42-f5551fab6ccb" />